include/slb/functional.hpp

```diff
@@ -175,6 +176,82 @@ invoke(F&& f,
 }
 #endif
 
+// [func.not_fn], function template not_fn
+
+// We only enable the C++17 implementation under C++2a here to account for
+// P0704: "Fixing const-qualified pointers to members" and LWG2210: "INVOKE-ing
+// a pointer to member with a `reference_wrapper` as the object expression".
+
+#if __cpp_lib_not_fn /* C++17 */ && __cplusplus > 201703L /* C++2a */
+using std::not_fn;
+#else
+namespace detail {
+struct not_fn_tag {
+  explicit not_fn_tag() = default;
+};
+
+template <typename FD>
+class not_fn_result {
+  FD fd;
+
+public:
+  template <typename F>
+  not_fn_result(not_fn_tag, F&& f) : fd(std::forward<F>(f)) {}
+
+  not_fn_result(not_fn_result&&) = default;
+  not_fn_result(not_fn_result const&) = default;
+
+  template <typename... Args>
+      auto operator()(Args&&... args) &
+      noexcept(noexcept(!detail::invoke(std::declval<FD&>(),
```

Comment:
http://eel.is/c++draft/func.not_fn doesn't seem to mention these `noexcept` conditions...
Is there maybe a relevant LWG issue regarding this?

Reply:
Not that I know, LWG doesn't generally do conditional `noexcept` other than move/swap. Implementations are free to add them.

Reply:
> Implementations are free to add them.
👍